### PR TITLE
Simplify UI scaler

### PR DESCRIPTION
**Pull request by kianzarrin**

### CODE REVIEW

File: TLM/TLM/U/UIScaler.cs [hunk 1]
```diff
@@ -8,51 +8,28 @@ namespace TrafficManager.U {
...
-            }
-        }
-
-        private static float BaseResolutionY {
-            get {
-                if (TryGetScreenResolution(out Vector2 resolution)) {
-                    // always 1080f. But we keep this code for the sake of future proofing
-                    return resolution.y;
-                }
+        private static float BaseResolutionX => UIView.GetAView().GetScreenResolution().x;
```
Comment (DaEgi01) [inline TLM/TLM/U/UIScaler.cs:11]: I'd reduce GetAView() calls to a minimum.
Call it once and cache it.
The method searches every time for the first UIView even though the value never changes.
Also the method is implemented via LINQ and will cause heap allocations each time.

**Review by DaEgi01 — CHANGES_REQUESTED** [review]

**Review by kvakvs — APPROVED** [review]
Renaming config option will reset scale for everyone.

**Comment by DaEgi01** [comment]
> Renaming config option will reset scale for everyone.

I see, how do we handle that?
1. we don't care about the user?
2. we don't care about the code?
:D

**Comment by kvakvs** [comment]
> I see, how do we handle that?
>     1. we don't care about the user?
>     2. we don't care about the code?

Reasonable default

or keep old variable, give new variable default -1 for example, and if it is -1, we migrate value to new on the first load. This migration code can be removed in the next release.

**Comment by DaEgi01** [comment]
> > I see, how do we handle that?
> > 1. we don't care about the user?
> > 2. we don't care about the code?
> 
> Reasonable default for example 100% zoom
> 
> or keep old variable, give new variable default -1 for example, and if it is -1, we migrate value to new on the first load. This migration code can be removed in the next release.

To be honest, in that case I'd probably leave the variable name as is since its too much hastle for too little gain. I did not consider that when I proposed to rename the variable. But your solution also sounds fine.

**Review by DaEgi01 — APPROVED** [review]

**Review by DaEgi01 — APPROVED** [review]

**Review by krzychu124 — APPROVED** [review]
👍

**Pull request merged**